dashboard/app/models/pd/session.rb

```diff
@@ -78,11 +78,21 @@ def open_for_attendance?
   end
 
   def too_soon_for_attendance?
-    workshop.started_at.nil? || start - 12.hours > Time.zone.now
+    workshop.started_at.nil? || start.beginning_of_day > Time.zone.now
```

Comment:
I made this change so that when we say you can only mark yourself as attended on the day of the workshop, that it is actually true. If a workshop session starts after noon, you wouldn't be able to mark yourself as attended until 12 hours before the start date. Similarly, if the session starts early in the morning, you could mark yourself as attended before the day of the session. 

So what is `Time.zone.now` doing then? Is it not getting the "local" times?

Reply:
`Time.zone.now` is the "local" time on the front end server, which is UTC. That will be the same regardless of where in the world the client viewing the result is. Remember all this Ruby / Rails code runs on the server and is always UTC.

The session start and end times are not relative to anything, they're just times that the workshop organizers fill in. Originally it was not used for anything automated. We have two main choices:

 1. Estimate like before. Instead of exactly beginning / end of day do a time span that roughly covers what we want across the US. 
 2. Be more sophisticated with session times and try to adjust based on the workshop location.

I still think using the actual beginning/end of day will be extra confusing and cause issues. For instance, midnight UTC is 5pm Seattle time, so the "next day" will start at 5pm here when we're likely to still have valid open workshops. Similarly the next day's workshop will start at 5pm the previous day.

Reply:
Okay that makes sense, I didn't realize it was all in UTC even when you use `.zone`.

I think estimating for beginning of day like it currently is will be fine. It did seem important that attendance ends around midnight of the day of the session, however. Could we set the cut off time to be 7am UST? That way it's midnight for PST and 3am for EST?

Reply:
It's only because the time zone of our front end servers is UTC. That same code on your localhost will use your time zone, as expected (at least whatever your computer is configured to).

That estimate SGTM. @tanyaparker ?

Reply:
Closing at midnight Pacific/3am Eastern seems fine to me. I'd rather err on keeping it open longer past midnight, than closing it too soon like 5pm.